lib/buckets/common.js

```diff
@@ -156,29 +156,148 @@ function _list(type, req, bucket_id) {
 
     assert.uuid(owner, 'owner');
 
-    // Validate optional delimiter
-        if (delimiter && delimiter.length > 1) {
+    // Prefix validation
+    if (prefix) {
+        // Existing null byte check
+        if (nullRegex.test(prefix)) {
+            process.nextTick(function () {
+                ee.emit('error', new InvalidParameterError('prefix',
+                    'Prefix parameter contains null bytes'));
+            });
+            return (ee);
+        }
+
+        // CRLF injection check
+        if (/[\r\n]/.test(prefix)) {
```

Comment:
All  the inline regex constructors in this file should be moved to the top of the module so that the regular expressions are not re-instantiated on each call to `_list` (like was done for `nullRegex`, `resemblesIpRegex`, `bucketRegex`)